config/configcompression/compressiontype.go

```diff
@@ -5,31 +5,56 @@ package configcompression // import "go.opentelemetry.io/collector/config/config
 
 import "fmt"
 
-type CompressionType string
+// CompressionType represents a compression method
+// Deprecated [0.94.0]: use Type instead.
+type CompressionType = Type
+
+// Type represents a compression method
+type Type string
 
 const (
-	Gzip    CompressionType = "gzip"
-	Zlib    CompressionType = "zlib"
+	TypeGzip    Type = "gzip"
+	TypeZlib    Type = "zlib"
+	TypeDeflate Type = "deflate"
+	TypeSnappy  Type = "snappy"
+	TypeZstd    Type = "zstd"
+	typeNone    Type = "none"
+	typeEmpty   Type = ""
+
+	// Gzip
+	// Deprecated [0.94.0]: use TypeGzip instead.
+	Gzip CompressionType = "gzip"
+
+	// Zlib
+	// Deprecated [0.94.0]: use TypeZlib instead.
+	Zlib CompressionType = "zlib"
+
+	// Deflate
+	// Deprecated [0.94.0]: use TypeDeflate instead.
 	Deflate CompressionType = "deflate"
-	Snappy  CompressionType = "snappy"
-	Zstd    CompressionType = "zstd"
-	none    CompressionType = "none"
-	empty   CompressionType = ""
+
+	// Snappy
+	// Deprecated [0.94.0]: use TypeSnappy instead.
+	Snappy CompressionType = "snappy"
+
+	// Zstd
+	// Deprecated [0.94.0]: use TypeZstd instead.
+	Zstd CompressionType = "zstd"
 )
 
-func IsCompressed(compressionType CompressionType) bool {
-	return compressionType != empty && compressionType != none
+func IsCompressed(compressionType Type) bool {
+	return compressionType != typeEmpty && compressionType != typeNone
 }
 
-func (ct *CompressionType) UnmarshalText(in []byte) error {
-	switch typ := CompressionType(in); typ {
-	case Gzip,
-		Zlib,
-		Deflate,
-		Snappy,
-		Zstd,
-		none,
-		empty:
+func (ct *Type) UnmarshalText(in []byte) error {
+	switch typ := Type(in); typ {
```

Comment:
nit: Sorry, but my mind goes crazy when I see a switch used like an if :)) Can we fix this probably in a separate PR :)

Reply:
Are you objecting to `typ := Type(in); typ` or the switch statement in general?

Reply:
In general I don't like `switch` statements with only 2 branches, the are if's no? :)) But is a nit and not a blocker for sure.